enum.c

```diff
@@ -3796,24 +3796,78 @@ slicebefore_i(RB_BLOCK_CALL_FUNC_ARGLIST(yielder, enumerator))
 
 /*
  *  call-seq:
- *     enum.slice_before(pattern)                             -> an_enumerator
- *     enum.slice_before { |elt| bool }                       -> an_enumerator
+ *    slice_before(pattern)       -> enumerator
+ *    slice_before {|array| ... } -> enumerator
  *
- *  Creates an enumerator for each chunked elements.
- *  The beginnings of chunks are defined by _pattern_ and the block.
-
- *  If <code>_pattern_ === _elt_</code> returns <code>true</code> or the block
- *  returns <code>true</code> for the element, the element is beginning of a
- *  chunk.
-
- *  The <code>===</code> and _block_ is called from the first element to the last
- *  element of _enum_.  The result for the first element is ignored.
-
- *  The result enumerator yields the chunked elements as an array.
- *  So +each+ method can be called as follows:
+ *  Returns an Enumerator.
  *
- *    enum.slice_before(pattern).each { |ary| ... }
- *    enum.slice_before { |elt| bool }.each { |ary| ... }
+ *  With argument +pattern+, returns an enumerator that uses the pattern
+ *  to partition elements into arrays.
+ *  An element begins a new slice if <tt>element === pattern</tt>
+ *  (or if it is the first element).
+ *
+ *  Example with a range and integer pattern:
+ *
+ *   e = (1..15).slice_before(6) # => #<Enumerator: ...>
+ *   e.each {|array| p array }
+ *
+ *  Output:
+ *
+ *    [1, 2, 3, 4, 5]
+ *    [6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+ *
+ *  Example with an array and string pattern:
```

Comment:
I don't think we need 4 examples of pattern argument usage, as they are more of an example of differences in `===` behavior than `slice_before` behavior.  A single example should suffice, pick your favorite.

Reply:
Done.